tests/test_updatediff.py

```diff
@@ -983,3 +984,85 @@ def function_two():
             print("Previous line lied.")
         """
     )
+
+
+@pytest.mark.parametrize(
+    "filename",
+    [
+        "README.md",
+        "spa ces",
+        'qu`o"tes',
+        "m4â4ñ4a",
+    ],
+)
+def test_conflicted_files_are_marked_unmerged(
+    tmp_path_factory: pytest.TempPathFactory,
+    filename: str,
+) -> None:
+    # Template in v1 has a file with a single line;
+    # in v2 it changes that line.
+    # Meanwhile, downstream project appended contents to the first line.
+    src, dst = map(tmp_path_factory.mktemp, ("src", "dst"))
+
+    # First, create the template with an initial file
+    build_file_tree(
+        {
+            (src / filename): "upstream version 1",
+            (src / "{{_copier_conf.answers_file}}.jinja"): (
+                "{{_copier_answers|to_nice_yaml}}"
+            ),
+        }
+    )
+    with local.cwd(src):
+        git_init("hello template")
+        git("tag", "v1")
+
+    # Generate the project a first time, assert the file exists
+    run_copy(str(src), dst, defaults=True, overwrite=True)
+    assert (dst / filename).exists()
+    assert "_commit: v1" in (dst / ".copier-answers.yml").read_text()
+
+    # Start versioning the generated project
+    with local.cwd(dst):
+        git_init("hello project")
+
+        # After first commit, change the file, commit again
+        Path(filename).write_text("upstream version 1 + downstream")
+        git("commit", "-am", "updated file")
+
+    # Now change the template
+    with local.cwd(src):
+        # Update the file
+        Path(filename).write_text("upstream version 2")
+
+        # Commit the changes
+        git("add", ".", "-A")
+        git("commit", "-m", "change line in file")
+        git("tag", "v2")
+
+    # Finally, update the generated project
+    run_update(dst_path=dst, defaults=True, overwrite=True, conflict="inline")
+    assert "_commit: v2" in (dst / ".copier-answers.yml").read_text()
+
+    # Assert that the file still exists, has inline markers,
+    # and is reported as "unmerged" by Git.
+    assert (dst / filename).exists()
+
+    expected_contents = dedent(
+        """\
+        <<<<<<< before updating
+        upstream version 1 + downstream
+        =======
+        upstream version 2
+        >>>>>>> after updating
+        """
+    )
+    assert (dst / filename).read_text().splitlines() == expected_contents.splitlines()
+    assert not (dst / f"{filename}.rej").exists()
+
+    with local.cwd(dst):
+        lines = git("ls-files", "--stage", filename).strip().splitlines()
+    assert len(lines) == 3
+    assert " 1\t" in lines[0]
+    assert " 2\t" in lines[1]
+    assert " 3\t" in lines[2]
```

Comment:
We could also check the output of `git status` instead of `ls-files --stage`.

Reply:
I think it would make total sense. You can check `git status --porcelain=v1` (or v2) so it's more parseable.

The output looks like this:

```
gitpod /tmp/pytest-of-gitpod/pytest-0/popen-gw0/dst0 (main) $ git status
On branch main
Unmerged paths:
  (use "git restore --staged <file>..." to unstage)
  (use "git add <file>..." to mark resolution)
        both modified:   "qu`o\"tes"

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   .copier-answers.yml

no changes added to commit (use "git add" and/or "git commit -a")

gitpod /tmp/pytest-of-gitpod/pytest-0/popen-gw0/dst0 (main) $ git status --porcelain=v1
 M .copier-answers.yml
UU "qu`o\"tes"

gitpod /tmp/pytest-of-gitpod/pytest-0/popen-gw0/dst0 (main) $ git status --porcelain=v2
1 .M N... 100644 100644 100644 8ba1f2648a04776ae6877dbc8d7aac5bea9e997c 8ba1f2648a04776ae6877dbc8d7aac5bea9e997c .copier-answers.yml
u UU N... 100644 100644 100644 100644 f163f4b09aad0c495f8f58112ea4aabc9900afe7 f163f4b09aad0c495f8f58112ea4aabc9900afe7 f163f4b09aad0c495f8f58112ea4aabc9900afe7 "qu`o\"tes"
```

... so, I guess `--porcelain=v1` will be the easiest one for a test.

Reply:
Done!